src/main/java/com/slack/kaldb/config/KaldbConfig.java

```diff
@@ -26,7 +27,19 @@
       throws InvalidProtocolBufferException {
     KaldbConfigs.KaldbConfig.Builder kaldbConfigBuilder = KaldbConfigs.KaldbConfig.newBuilder();
     JsonFormat.parser().ignoringUnknownFields().merge(jsonStr, kaldbConfigBuilder);
-    return kaldbConfigBuilder.build();
+    KaldbConfigs.KaldbConfig kaldbConfig = kaldbConfigBuilder.build();
+    validateConfig(kaldbConfig);
+    return kaldbConfig;
+  }
+
+  public static void validateConfig(KaldbConfigs.KaldbConfig kaldbConfig) {
+    // We don't need further checks for node roles since JSON parsing will throw away roles not part
```

Comment:
Let's use a pre-condition instead for this check. 
 https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html

Does the validation need to be here? I think it may be better to move it closer to main in KalDb.java.

Reply:
if we move it to kalDB how do you have any recommendations on how to unit test it?

Currently KaldbcConfigTest has a lot of tests - How will we migrate those to use Kaldb.java?